### PR TITLE
Fix #7082 - Site:GroundTemperature:Undisturbed:FiniteDifference results in crash

### LINKED ISSUE #7082: Site:GroundTemperature:Undisturbed:FiniteDifference results in crash
Issue overview
--------------
Running EnergyPlus with a Pipe:Underground object where the 'Type of Undisturbed Ground Temperature Object' field is "Site:GroundTemperature:Undisturbed:FiniteDifference" results in an undefined crash.

The remaining types ("Site:GroundTemperature:Undisturbed:KusudaAchenbach" and "Site:GroundTemperature:Undisturbed:Xing") work correctly.

The crash occurred in a user-defined file and also in the Pipe:Underground example file: PipeHeatTransfer_Underground.idf

### Details
Some additional details for this issue (if relevant):
 - Windows 10 x64
 - Version of EnergyPlus: 9.0.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added: https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/7000s/7082/7082_PipeHeatTransfer_Underground_mod.idf#L3167:L3195
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

Fix #7082 - Site:GroundTemperature:Undisturbed:FiniteDifference results in crash

Added unit test and checked that https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/7000s/7082/7082_PipeHeatTransfer_Underground_mod.idf crashes without and runs with it.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Myoldmopar** [comment]
@mitchute and/or @jasondegraw do either of you have any thoughts on these changes?  I was caught off-guard seeing that the fix was actually in the weather manager, not in the ground temperature model.  I am sure it's the right fix, but I think both of you have been in this code for different reasons lately and I'd love to hear your thoughts.

File: tst/EnergyPlus/unit/FiniteDifferenceGroundTemperatureModel.unit.cc [hunk 1]
```diff
@@ -51,15 +51,20 @@
...
+#include <DataEnvironment.hh>
+#include <SimulationManager.hh>
+#include <ElectricPowerServiceManager.hh>
+#include <ConfiguredFunctions.hh>
+
 #include "Fixtures/EnergyPlusFixture.hh"
 
 using namespace EnergyPlus;
-using namespace EnergyPlus::GroundTemperatureManager;
+//using namespace EnergyPlus::GroundTemperatureManager;
```
Comment (mitchute) [inline tst/EnergyPlus/unit/FiniteDifferenceGroundTemperatureModel.unit.cc]: Remove?

**Comment by mitchute** [comment]
@Myoldmopar I'm not surprised by this. I think when the F-D ground temp model went in, reading the weather file in entirety before the full simulation was a bit more finicky than it is now. I haven't kept up on how that capability has been upgraded since then, but I'm not surprised that this needed to be pulled up to the current methods.

**Comment by jasondegraw** [comment]
@Myoldmopar I'm not surprised either. The weather file reading still seems pretty finicky to me (and seems to always happen at least one more time than expected). This is why we need a better weather file facility, hope we can get that going soon(ish).

**Comment by Myoldmopar** [comment]
I'm not super pumped about this, but it's not your stuff I'm struggling with on this one @jmarrec. With this in mind, I'm just going to merge this in.  It fixes the issue, and doesn't break anything else.  Thanks for the fix.

**Pull request merged**